pkg/tests/end_to_end_tests/config_dataset_test.go

```diff
@@ -0,0 +1,47 @@
+package end_to_end_tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/jackc/pgx/v4/pgxpool"
+	"github.com/stretchr/testify/require"
+	"github.com/timescale/promscale/pkg/dataset"
+)
+
+func TestDatasetConfigApply(t *testing.T) {
+
+	withDB(t, *testDatabase, func(dbOwner *pgxpool.Pool, t testing.TB) {
+		conn, err := dbOwner.Acquire(context.Background())
+		require.NoError(t, err)
+		defer conn.Release()
+
+		pgxConn := conn.Conn()
+		require.Equal(t, getDefaultChunkInterval(t, pgxConn), 8*time.Hour)
+
+		cfg := dataset.Config{Metrics: dataset.Metrics{ChunkInterval: 4 * time.Hour}}
+
+		err = cfg.Apply(pgxConn)
+		require.NoError(t, err)
+
+		require.Equal(t, getDefaultChunkInterval(t, pgxConn), 4*time.Hour)
+
+		// Set to default if chunk interval is not specified.
+		cfg.Metrics.ChunkInterval = 0
```

Comment:
How can it not be specified in a test case?

Reply:
If a user runs the dataset config option but doesn't specify chunk interval, it will be set to zero and hence updated to the default value. Its a little strange now because there is only one config option but in the future when there are many, it will make more sense.